gateway/src/apicast/policy/3scale_batcher/3scale_batcher.lua

```diff
@@ -8,16 +8,27 @@ local reporter = require('reporter')
 local Transaction = require('transaction')
 local http_ng_resty = require('resty.http_ng.backend.resty')
 local semaphore = require('ngx.semaphore')
+local TimerTask = require('resty.concurrent.timer_task')
 
 local ipairs = ipairs
 
 local default_auths_ttl = 10
 local default_batch_reports_seconds = 10
 
-local _M = policy.new('Caching policy')
+local _M, mt = policy.new('Caching policy')
 
 local new = _M.new
 
+mt.__gc = function(self)
+  -- Instances of this policy are garbage-collected when the config is
+  -- reloaded. We need to ensure that the TimerTask instance schedules another
+  -- run before that so we do not leave any pending reports.
+
+  if self.timer_task then
+    self.timer_task:schedule_one(1)
```

Comment:
Why delay 1 and not 0 ?

Reply:
:)

From the point of view of the policy, I think it does not matter. However, busted crashes with a segfault when this is 0. Not sure why exactly. It might be related to how busted collects the garbage after all the tests. It does not crash with a delay of 0.1.

Reply:
Ok, then this is definitely worth a comment :-D

Reply:
In the end I changed this and implemented it as we discussed off-line: https://github.com/3scale/apicast/pull/786/commits/ba883c101ee8990af50cb56cd52784d74df00743